blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,363 @@
+---
+title: Fobnail Token - example use case
+abstract: 'This phase focused on using Fobnail in a real-life use case, namely
+          using it to access LUKS2 decryption key if and only if the PCR
+          measurements are valid'
+cover: /covers/usb_token.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2022-12-22
+archives: "2022"
+
+tags:
+  - fobnail
+  - tpm
+  - attestation
+  - linux
+categories:
+  - Security
+
+---
+
+# About the Fobnail project
+
+Fobnail is a project that aims to provide a reference architecture for building
+offline integrity measurement verifiers on the USB device (Fobnail Token) and
+attesters running in Dynamically Launched Measured Environments (DLME). It
+allows the Fobnail owner to verify the trustworthiness of the running system
+before performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://blog.3mdeb.com/tags/fobnail/) tag.
+
+# Scope of current phase
+
+This phase focused on using Fobnail in a real-life use case, namely using it to
+access LUKS2 decryption key if and only if the PCR measurements are valid. In
+this post only high-level usage will be described, if you want to know what
+happens under the hood or want to modify it to your needs see
+[documentation](https://fobnail.3mdeb.com/examples/disk_encryption/).
```

Comment:
What if PCRs will change because of firmware update?

Reply:
@krystian-hebel but updating firmware is valid user scenario. We don't want users to shot themselves in foot, or maybe other way: we are ok if one will do that, but we should inform about the problem and potentially think about providing recovery path. Any security approach needs protection, detection and recovery mechanism otherwise it is not really useful.

Reply:
It really depends on threat model. I will add instructions for what to do with FW updates to documentation, this post is probably already long enough.

Reply:
https://github.com/fobnail/docs/pull/48